addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1041,6 +1041,122 @@ const ColorpickerUserValueWidget = SelectUserValueWidget.extend({
+        this.editImageButton.append(editIconEl);
+        this.editImageButton.addEventListener('click', (ev) => this._onEditImage(ev));
+
+        this.removeImageButton = document.createElement('we-button');
+        this.removeImageButton.classList.add('fa', 'fa-fw', 'fa-times');
+        this.removeImageButton.title = _t("Remove image");
+        this.removeImageButton.addEventListener('click', (ev) => this._onRemoveImage(ev));
+
+        return this.containerEl.append(_buildRowElement('', {
+            childNodes: [this.editImageButton, this.removeImageButton],
+        }));
+    },
+    /**
+     * @override
+     */
+    getMethodsParams: function (methodName) {
+        return _.extend({isVideo: this.isVideo}, this._super(...arguments));
+    },
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    updateUI: async function () {
+        await this._super(...arguments);
+        this.removeImageButton.classList.toggle('d-none', !this.isActive());
+        this.editBgTextEl.innerText = this.isActive() ? this._getDisplayName() : this.defaultText;
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Computes the name to display in the edit button.
+     *
+     * @private
+     */
+    _getDisplayName: function () {
+        if (this.isVideo) {
+            return this._value;
+        }
+        return this._value.substring(this._value.lastIndexOf('/') + 1).split(/[?#]/)[0];
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Called when the edit image button is clicked.
+     *
+     * @private
+     */
+    _onEditImage: function (ev) {
+        // Need a dummy element for the media dialog to modify.
+        const dummyEl = document.createElement(this.isVideo ? 'iframe' : 'img');
+        dummyEl.src = this._value;
+        if (this.isVideo) {
+            // Allows the mediaDialog to select the video tab immediately.
+            dummyEl.classList.add('media_iframe_video');
+        }
+        const mediaDialog = new weWidgets.MediaDialog(this, {
+            noIcons: true,
+            noDocuments: true,
+            noVideos: !this.allowVideos,
+            isForBgVideo: true,
+        }, dummyEl).open();
+        mediaDialog.on('save', this, data => {
+            if (data.bgVideoSrc) {
+                this.setValue(data.bgVideoSrc);
+                this.isVideo = true;
+            } else {
+                // Accessing the value directly through img.src converts the url to absolute
+                this.setValue(dummyEl.getAttribute('src'));
```

Comment:
We probably do not want it to absolute though?

Reply:
That's the point. If you use `dummyEl.src`, you get a complete, absolute URL (with protocol and everything). Using `getAttribute` allows me to get the URL as it is written in the src attribute, which is useful in some cases (eg: comparing it with the relative URLs stored in DB to find the corresponding attachment)

Reply:
Ok, the comment in fact confused me as instead of using `data` you used the `dummyEl` image to read the src. I did not get that it was the way you were reading it that changed.

Reply:
I made the comment a lot more explicit so that there is no misunderstanding.